### DISCUSSION TITLE
VirtualList itemSize

### DISCUSSION THREAD

**Discussion by china-powers** [NervJS/taro#9298] [Q & A] [answered]
长列表itemSize怎么转为rpx？
单项高度为200，会根据页面自动转为rpx，但是itemSize是以px为高度，导致两边高度不一致，如何解决？

**Comment by ZakaryCode** [comment, marked as answer]
可以看看文档这里哦

https://nervjs.github.io/taro/docs/size.html#api

**Reply by undeadfrost** [reply]
@ZakaryCode VirtualList width height itemSize这些都是固定px啊？样式里面的都转换成rpx，不同机型可以进行自动适配，可是这个组件怎么处理呢？这个文档没有说明这个问题。

**Reply by ZakaryCode** [reply]
这里没办法支持，但是有两个解决思路：
- 解开高度限制，组件会帮助你根据组件自适应高度；
- 如果希望统一高度，但又不想传入 px，可以手动计算，或者使用 API 获取对应节点高度传入。

**Reply by tiny-dust** [reply]
> 

解开高度限制，是`unlimitedSize`这个属性吗? 我试了一下，貌似没什么区别

**Comment by joket1999** [comment]
做一个骨架屏组件，骨架屏item的高度跟实际列表的item高度一样，然后获取item的rect得到height，再付给虚拟列表的itemSize

**Reply by jiajiaxing1** [reply]
做一个骨架屏，本身也需要先获取到实际列表item的高度，但是实际列表的item的高度本身就是个动态的不确定的高度，根据内容撑开的
